include/dlaf/eigensolver/band_to_tridiag/mc.h

```diff
@@ -467,9 +468,11 @@ TridiagResult<T, Device::CPU> BandToTridiag<Backend::MC, D, T>::call_L(
 
       const GlobalElementIndex index_v((sweep / b + step) * b, sweep);
 
-      pika::dataflow(pika::launch::sync, unwrapping(store_tau_v), w_pipeline(),
-                     mat_v(dist_v.globalTileIndex(index_v)), dist_v.tileElementIndex(index_v));
-      deps[step] = pika::dataflow(executor_hp, unwrapping(cont_sweep), w_pipeline(), deps[dep_index]);
+      dlaf::internal::whenAllLift(w_pipeline(), mat_v.readwrite_sender(dist_v.globalTileIndex(index_v)),
+                                  dist_v.tileElementIndex(index_v)) |
+          ex::then(store_tau_v) | ex::start_detached();
+      deps[step] = dlaf::internal::whenAllLift(w_pipeline(), deps[dep_index]) |
+                   dlaf::internal::transform(policy_hp, cont_sweep) | ex::make_future();
```

Comment:
The reason is the same as in https://github.com/eth-cscs/DLA-Future/pull/475#discussion_r841616786, i.e. only to keep the `launch::sync` behaviour unchanged.

Reply:
IMO he preferred it because it was closer to the original code. Indeed, you can spot the `dataflow(launch::sync, ...)` in the old code, which means that we wanted to have a continuation, which is explicit in the new code with the `then`.

I anticipated @msimberg 😄 just for checking my understanding.